core/core.go

```diff
@@ -422,8 +422,14 @@ func downloadBazelIfNecessary(version string, baseDirectory string, repos *Repos
 	}
 
 	var tmpDestPath string
-	if url := GetEnvOrConfig(BaseURLEnv); url != "" {
-		tmpDestPath, err = repos.DownloadFromBaseURL(url, version, destDir, tmpDestFile)
+	baseURL := GetEnvOrConfig(BaseURLEnv)
+	formatURL := GetEnvOrConfig(FormatURLEnv)
+	if baseURL != "" && formatURL != "" {
+		return "", fmt.Errorf("cannot set %s and %s at once", BaseURLEnv, FormatURLEnv)
+	} else if formatURL != "" {
+		tmpDestPath, err = repos.DownloadFromFormatURL(formatURL, version, destDir, tmpDestFile)
```

Comment:
I acknowledge that adding this feels very strange here because we have an input `downloader` function that should encapsulate this. *However* please note that the previous code was already hacking its way to handle the base URL case in the same way, so I just followed existing practice.

I think that this could be generalized to inject these behaviors via `downloader` functions... but I'd rather investigate that separately because there are lots of subtle consequences...